ComposeSamples/NavigationRail/src/main/java/com/microsoft/device/display/samples/navigationrail/ui/view/Navigation.kt

```diff
@@ -0,0 +1,180 @@
+val navDestinations = GallerySections.values()
+
+enum class GallerySections(
+    @StringRes val title: Int,
+    @DrawableRes val icon: Int,
+    val route: String,
+    val list: List<Image>,
+    @DrawableRes val placeholderImage: Int,
+) {
+    PLANTS(
+        R.string.plants,
+        R.drawable.plant_icon,
+        "plants",
+        DataProvider.plantList,
+        R.drawable.plants_placeholder
+    ),
+    BIRDS(
+        R.string.birds,
+        R.drawable.bird_icon,
+        "birds",
+        DataProvider.birdList,
+        R.drawable.birds_placeholder
+    ),
+    ANIMALS(
+        R.string.animals,
+        R.drawable.animal_icon,
+        "animals",
+        DataProvider.animalList,
+        R.drawable.animals_placeholder
+    ),
+    LAKES(
+        R.string.lakes,
+        R.drawable.lake_icon,
+        "lakes",
+        DataProvider.lakeList,
+        R.drawable.lakes_placeholder
+    ),
+    ROCKS(
+        R.string.rocks,
+        R.drawable.rock_icon,
+        "rocks",
+        DataProvider.rockList,
+        R.drawable.rocks_placeholder
+    )
+}
+
+/**
+ * Build nav graph with the different galleries as destinations
+ */
+@ExperimentalFoundationApi
+fun NavGraphBuilder.addGalleryGraph(
+    currentImageId: Int?,
+    onImageSelected: (Int) -> Unit,
+    showItemView: Boolean,
+    horizontalPadding: Dp
+) {
+    navDestinations.forEach { section ->
+        composable(section.route) {
+            Scaffold(
+                topBar = { GalleryTopBar(section.route, horizontalPadding) }
+            ) {
+                GalleryOrItemView(
+                    galleryList = section.list,
+                    currentImageId = currentImageId,
+                    onImageSelected = { id -> onImageSelected(id) },
+                    showItemView = showItemView,
+                    horizontalPadding = horizontalPadding,
+                )
+            }
+        }
+    }
+}
+
+/**
+ * Show the NavHost with the gallery composables, surrounded by a top bar and the appropriate nav
+ * component (BottomNavigation or NavigationRail)
+ */
+@ExperimentalAnimationApi
+@ExperimentalMaterialApi
+@ExperimentalFoundationApi
+@Composable
+fun ShowWithNav(
+    isDualScreen: Boolean,
+    isDualPortrait: Boolean,
+    imageId: Int?,
+    updateImageId: (Int?) -> Unit,
+    currentRoute: String,
+    updateRoute: (String) -> Unit
+) {
+    val navController = rememberNavController()
+
+    // Use navigation rail when dual screen (more space), otherwise use bottom navigation
+    Scaffold(
+        bottomBar = {
+            if (!isDualScreen)
+                GalleryBottomNav(navController, navDestinations, updateImageId, updateRoute)
+        },
+    ) { paddingValues ->
+        Row(Modifier.padding(paddingValues)) {
+            if (isDualScreen)
+                GalleryNavRail(navController, navDestinations, updateImageId, updateRoute)
+            NavHost(
+                modifier = Modifier.onGloballyPositioned {
```

Comment:
Why do we need this? Is there a scenario in which the route is not `currentRoute`?